flow/Platform.cpp

```diff
@@ -1796,7 +1796,55 @@ bool createDirectory( std::string const& directory ) {
 
 }; // namespace platform
 
+const uint8_t separatorChar = CANONICAL_PATH_SEPARATOR;
+StringRef separator(&separatorChar, 1);
+StringRef dotdot = LiteralStringRef("..");
+
+std::string cleanPath(std::string const &path) {
+	std::vector<StringRef> parts;
+	std::vector<StringRef> finalParts;
+	bool absolute = !path.empty() && path[0] == CANONICAL_PATH_SEPARATOR;
+
+	StringRef p(path);
+
+	while(p.size() != 0) {
+		StringRef part = p.eat(separator);
+		if(part.size() == 0 || (part.size() == 1 && part[0] == '.'))
+			continue;
+		if(part == dotdot) {
+			if(!finalParts.empty() && finalParts.back() != dotdot) {
+				finalParts.pop_back();
+				continue;
+			}
+			if(absolute) {
```

Comment:
Actually doing `..` when one is already in `/` is a no-op. Maybe we should simply do no-op too?

Reply:
I actually did not realize that, that sounds good.

Reply:
Hmm does this also mean that parentDirectory of root should be root?

Reply:
Good thought. `dirname` in *nix system acts somewhat the same as getting the parent directory of a given path, and it does so, i.e. parent of root is still root.